packages/core/src/select.js

```diff
@@ -11,21 +13,30 @@ type PluralForms = {
 
 type PluralProps = {
   value: number,
-  offset?: number
+  offset?: number,
+  locales?: any,
```

Comment:
I'm pretty sure this type should be:
```
locales?: ?string | string[],
```
but flow is complaining that `null or undefined` and `array` are incompatible with `string`. I don't see where the error is because locales is declared with the same type everywhere `?string | string[]`.

Reply:
Why is it array? It always has to be just a single locale, right?

Reply:
Intl accepts a string or an array to let the user provide a fallback in case the desired locale is not available. The first item is the number one priority, the second item is number two priority...

Reply:
It seems that Flow can't handle default values inside object destructuring for union types. It fails also when I set `format` type (formerly `options`) to `NumberFormat` with default value `{}`. I moved default value declaration inside function (which is what Babel does anyway) and linting works fine.